File: xendit_cards_session/lib/src/utils/credit_card_util.dart
/// Utility class for credit card validation
class CreditCardUtil {
  /// Validates if a credit card number is valid
  /// 
  /// Currently just checks if the number is not empty
  /// TODO: Implement full validation logic including Luhn algorithm
  static bool isCreditCardNumberValid(String cardNumber) {
    return cardNumber.trim().isNotEmpty;
  }

  /// Validates if a credit card expiration date is valid
  /// 
  /// Checks if month is between 1-12 and year is not in the past
  static bool isCreditCardExpirationDateValid(String month, String year) {
    final monthInt = int.tryParse(month);
    final yearInt = int.tryParse(year);
    
    if (monthInt == null || yearInt == null) {
      return false;
    }
    
    // Check if month is valid
    if (monthInt < 1 || monthInt > 12) {
      return false;
    }
    
    // Get current date for comparison
    final now = DateTime.now();
    final currentYear = now.year;
    
    // Check if year is valid (not in the past)
    return yearInt >= currentYear;
  }

  /// Validates if a credit card CVN is valid
  /// 
  /// CVN is optional, but if provided must be 3-4 digits
  static bool isCreditCardCVNValid(String? cvn) {
    if (cvn == null) return true; // CVN is optional
    if (cvn.trim().isEmpty) return false;
    
    final cvnDigits = cvn.trim();
    return cvnDigits.length >= 3 && 
           cvnDigits.length <= 4 && 
           cvnDigits.codeUnits.every((c) => c >= 48 && c <= 57); // Check if all characters are digits
  }
} 

File: android-sdk/app/src/main/java/com/cards/session/cardsession/sdk/CreditCardUtil.kt
package com.cardsession.sdk

/**
 * Utility class for credit card validation
 * TODO Implement all of https://github.com/xendit/xendit.js/blob/26106eef2939870058ee6c219ac837d4c45b7fcf/src/utils/credit_card_util.js
 */
object CreditCardUtil {
  fun isCreditCardNumberValid(cardNumber: String): Boolean {
    return cardNumber.isNotBlank()
  }

  fun isCreditCardExpirationDateValid(month: String, year: String): Boolean {
    val monthInt = month.toIntOrNull() ?: return false
    val yearInt = year.toIntOrNull() ?: return false

    return monthInt in 1..12 && yearInt >= 2024
  }

  fun isCreditCardCVNValid(cvn: String?): Boolean {
    if (cvn == null) return true // CVN is optional
    if (cvn.isBlank()) return false
    val cvnDigits = cvn.trim()
    return cvnDigits.length in 3..4 && cvnDigits.all { it.isDigit() }
  }
} 